src/Riok.Mapperly/Descriptors/SymbolAccessor.cs

```diff
@@ -370,6 +370,85 @@ internal bool TryFindMemberPath(ITypeSymbol type, StringMemberPath path, [NotNul
         return false;
     }
 
+    /// <summary>
+    ///     Checks that the specified method returns a type that can be assigned to the specified result type,
+    ///     and that the specified arguments can be passed to the method.
+    ///     The check takes into account the possibility of assignment to a method with an argument marked with the <see langword="params"/> keyword
+    /// </summary>
+    /// <param name="method">Method for validate</param>
+    /// <param name="returnType">Target return type</param>
+    /// <param name="argTypes">Target method arguments</param>
+    /// <returns></returns>
+    internal bool ValidateSignature(IMethodSymbol method, ITypeSymbol returnType, params ITypeSymbol[] argTypes)
+    {
+        if (!CanAssign(method.ReturnType, returnType))
+        {
+            return false;
+        }
+
+        for (var i = 0; i < method.Parameters.Length; i++)
+        {
+            if (method.Parameters[i] is not { IsParams: true } isParamsParameter)
+            {
+                if (!CanAssign(argTypes[i], method.Parameters[i].Type))
+                {
+                    return false;
+                }
```

Comment:
Nit: an early continue could reduce the nesting level for the rest of the for loop body. Maybe it could even make sense to extract this into another method `ValidateParamater`